### DISCUSSION TITLE
not loading battery icon

### DISCUSSION THREAD

**Discussion by ankur-gupta-29** [starship/starship#6250] [Q&A]
i am using Gruvbox Rainbow . so i am edit startship.toml and add battery line

[battery]
full_symbol = '🔋 '
charging_symbol = '⚡️ '
discharging_symbol = '💀 '

after add these line no battery icon is showing why ?

**Comment by ghaaj** [comment]
If you are customizing the `format` option, you need to explicitly add its variable to `format` in order to display the enabled module in your prompt.
Try to update `starship.toml` as below:
```toml
"$schema" = 'https://starship.rs/config-schema.json'

format = """
[](color_orange)\
$os\
$username\
...
[](fg:color_bg3 bg:color_bg1)\
$time\
[ ](fg:color_bg1) $fill $battery\
$line_break$character"""

palette = 'gruvbox_dark'

[palettes.gruvbox_dark]
color_fg0 = '#fbf1c7'
color_bg1 = '#3c3...
```

**Reply by tornandbroken** [reply]
That Syntax is a little tricky
I has the same issue and I thought it has something to do with wayland but it works fine with it

starship.toml with modified "gruvbox-rainbow.toml" - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

`
right_format = """$battery"""
`
format = """

starship.toml with modified "gruvbox-rainbow.toml" - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

[battery]
full_symbol = '󰁹 '
charging_symbol = '󰂄 '
discharging_symbol = '󰂃 '
unknown_symbol = '󰁽 '
empty_symbol = '󰂎 '

[[battery.display]]
threshold = 20
style = "color_red"
[[battery.display]]
threshold = 100
style = "color_yellow"

In the standart starship toml file should look like that

starship.toml "empty" - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

`
right_format = """$battery"""
`

[battery]
full_symbol = '󰁹 '
charging_symbol = '󰂄 '
discharging_symbol = '󰂃 '
unknown_symbol = '󰁽 '
empty_symbol = '󰂎 '

[[battery.display]]
threshold = 20
style = "bold red"
[[battery.display]]
threshold = 100
style = "bold white"

starship.toml "empty" - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Nevertheless it works a little better with xorg

**Reply by MusaGillani** [reply]
Adding `$battery` _with these_ did the trick for me, maybe by default it only shows when % < 20 (which I didn't verify)
```toml
[[battery.display]]
threshold = 20
style = "bold red"
[[battery.display]]
threshold = 100
style = "bold white"
```
